Makefile

```diff
@@ -93,9 +98,17 @@ image: ## Build and push docker image to $registry
 
 .PHONY: deploy 
 deploy: ## Deploy cloud-api-adaptor using the operator, according to install/overlays/$(CLOUD_PROVIDER)/kustomization.yaml file.
+ifneq ($(CLOUD_PROVIDER),)
```

Comment:
what about having something like:
```
%:
        kubectl apply -f install/yamls/deploy.yaml
        kubectl apply -k install/overlays/$@

deploy: $(filter-out deploy, $(MAKECMDGOALS))
        @for i in $^; do ./$$i; done

```
this will also allow to use CLOUD_PROVIDERS instead of BUILD_PROVIDERS if proffered 
disclaimer: I'm not a makefile expert but IIUC it would allow to do: `make deploy aws `

Reply:
I'm not aware of the above syntax. Will give it a try.
Just to understand, you are suggesting using CLOUD_PROVIDERS instead of BUILD_PROVIDERS, which makes sense. 
However I don't understand how the user will provide a single provider just for deployment ?

Reply:
@snir911 are you suggesting running `deploy` target for all the providers that are part of CLOUD_PROVIDERS ?